paint/PAINT_Utils.cpp

```diff
@@ -17,14 +17,14 @@ DrawWindow * paint::utils::findDrawWindow(UIelement* control)
 	/* Implementation */
 
 	// Are we already at the DrawWindow?
-	if (auto dw = dynamic_cast<DrawWindow *>(control)) {
+	if (const auto dw = dynamic_cast<DrawWindow *>(control)) {
 		return dw;
 	}
 
 	// Are we the screen?
-	auto parent = control->getParent();
+	const auto parent = control->getParent();
 	if (!parent) {
-		auto screen = static_cast<Screen*>(control);
+		const auto screen = static_cast<Screen*>(control);
```

Comment:
It's entirely possible that we get our logic wrong, and somehow end up with a `UIelement` that is not a screen that has a null parent. So perhaps we could write this code as follows:

```
#if !defined(NDEBUG)
const auto screen = dynamic_cast<Screen *>(control);
#else
const auto screen = static_cast<Screen *>(control);
#endif
assert(screen);

```

Reply:
done